commit 6740e4cf278e7d59db2645ae8764d707dc515f9c (refs/changes/99/5399/12)
Author: Andy Fiddaman <omnios@citrus-it.co.uk>
Date:   2019-02-13T23:31:40+00:00 (8 months ago)
    
    OS-7528 update bhyve pkg manifest from OmniOS
    OS-7572 bhyve: stop shipping libvmm.so and libvmmapi.so
    Reviewed by: John Levon <john.levon@joyent.com>
    Reviewed by: Andy Fiddaman <omnios@citrus-it.co.uk>
    Approved by: John Levon <john.levon@joyent.com>

diff --git a/exception_lists/packaging b/exception_lists/packaging
index 0ab183dfa3..832a106ef7 100644
--- a/exception_lists/packaging
+++ b/exception_lists/packaging
@@ -1002,13 +1002,23 @@ usr/lib/llib-lsff
 usr/lib/llib-lsff.ln
 
 #
-# private bhyve header files
-#
-usr/include/bhyve.h
-usr/include/vmmapi.h
-usr/platform/i86pc/include/sys/vmm.h
-usr/platform/i86pc/include/sys/vmm_impl.h
-usr/platform/i86pc/include/sys/vmm_instruction_emul.h
+# private bhyve files
+#
+lib/amd64/libvmm.so			i386
+lib/amd64/libvmmapi.so			i386
+lib/amd64/llib-lvmmapi.ln		i386
+usr/include/bhyve.h			i386
+usr/include/libppt.h			i386
+usr/include/libvmm.h			i386
+usr/include/vmmapi.h			i386
+usr/lib/amd64/libppt.so			i386
+usr/lib/amd64/llib-lppt.ln		i386
+usr/lib/libppt.so			i386
+usr/lib/llib-lppt			i386
+usr/lib/llib-lppt.ln			i386
+usr/platform/i86pc/include/sys/vmm.h	i386
+usr/platform/i86pc/include/sys/vmm_impl.h	i386
+usr/platform/i86pc/include/sys/vmm_instruction_emul.h	i386
 
 #
 # libcustr is private
diff --git a/manifest b/manifest
index 6b591f7ccb..63b44a3d8e 100644
--- a/manifest
+++ b/manifest
@@ -1304,9 +1304,7 @@ s lib/amd64/libuuid.so=libuuid.so.1
 f lib/amd64/libuutil.so.1 0755 root bin
 s lib/amd64/libuutil.so=libuutil.so.1
 f lib/amd64/libvmm.so.1 0755 root bin
-s lib/amd64/libvmm.so=libvmm.so.1
 f lib/amd64/libvmmapi.so.1 0755 root bin
-s lib/amd64/libvmmapi.so=libvmmapi.so.1
 f lib/amd64/libw.so.1 0755 root bin
 s lib/amd64/libw.so=libw.so.1
 f lib/amd64/libxnet.so.1 0755 root bin
diff --git a/usr/src/pkg/manifests/system-bhyve-tests.mf b/usr/src/pkg/manifests/system-bhyve-tests.mf
new file mode 100644
index 0000000000..14586b5177
--- /dev/null
+++ b/usr/src/pkg/manifests/system-bhyve-tests.mf
@@ -0,0 +1,35 @@
+#
+# CDDL HEADER START
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright 2018 OmniOS Community Edition (OmniOSce) Association.
+#
+
+set name=pkg.fmri value=pkg:/system/bhyve/tests@$(PKGVERS)
+set name=pkg.description value="BSD hypervisor tests"
+set name=pkg.summary value="BSD hypervisor tests"
+set name=info.classification \
+    value=org.opensolaris.category.2008:System/Virtualization
+set name=variant.arch value=i386
+dir path=opt/bhyvetest
+dir path=opt/bhyvetest/bin
+dir path=opt/bhyvetest/tst
+dir path=opt/bhyvetest/tst/mevent
+file path=opt/bhyvetest/bin/bhyvetest mode=0555
+file path=opt/bhyvetest/tst/mevent/lists.delete.exe mode=0555
+file path=opt/bhyvetest/tst/mevent/read.disable.exe mode=0555
+file path=opt/bhyvetest/tst/mevent/read.pause.exe mode=0555
+file path=opt/bhyvetest/tst/mevent/read.requeue.exe mode=0555
+license lic_CDDL license=lic_CDDL
diff --git a/usr/src/pkg/manifests/system-bhyve.mf b/usr/src/pkg/manifests/system-bhyve.mf
index fe19fb21b4..002bef64cc 100644
--- a/usr/src/pkg/manifests/system-bhyve.mf
+++ b/usr/src/pkg/manifests/system-bhyve.mf
@@ -15,6 +15,7 @@
 
 #
 # Copyright 2018 Joyent, Inc.
+# Copyright 2019 OmniOS Community Edition (OmniOSce) Association.
 #
 
 #
@@ -30,31 +31,27 @@ set name=info.classification \
     value=org.opensolaris.category.2008:System/Virtualization
 set name=variant.arch value=i386
 dir path=kernel group=sys
-dir path=lib group=bin
-dir path=lib/$(ARCH64) group=bin
 dir path=usr group=sys
 dir path=usr/kernel/drv group=sys
 dir path=usr/kernel/drv/$(ARCH64) group=sys
-dir path=usr/lib group=bin
+dir path=usr/sbin
 dir path=usr/share
 dir path=usr/share/man
 dir path=usr/share/man/man1m
-dir path=usr/sbin
 driver name=ppt
 driver name=viona
 driver name=vmm
-file path=lib/$(ARCH64)/libvmmapi.so.1
-link path=lib/$(ARCH64)/libvmmapi.so target=./libvmmapi.so.1
 file path=usr/kernel/drv/$(ARCH64)/ppt
 file path=usr/kernel/drv/$(ARCH64)/viona
 file path=usr/kernel/drv/$(ARCH64)/vmm
 file path=usr/kernel/drv/ppt.conf
 file path=usr/kernel/drv/viona.conf
 file path=usr/kernel/drv/vmm.conf
-file path=usr/lib/libppt.so.1
-file path=usr/lib/$(ARCH64)/libppt.so.1
 file path=usr/sbin/bhyve mode=0555
 file path=usr/sbin/bhyvectl mode=0555
 file path=usr/sbin/pptadm mode=0555
 file path=usr/share/man/man1m/pptadm.1m
 license lic_CDDL license=lic_CDDL
+depend fmri=developer/acpi type=require
+depend fmri=system/bhyve/firmware type=require
+depend fmri=system/library/bhyve type=require
diff --git a/usr/src/pkg/manifests/system-library-bhyve.mf b/usr/src/pkg/manifests/system-library-bhyve.mf
new file mode 100644
index 0000000000..9af6e809cc
--- /dev/null
+++ b/usr/src/pkg/manifests/system-library-bhyve.mf
@@ -0,0 +1,33 @@
+#
+# CDDL HEADER START
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright 2019 OmniOS Community Edition (OmniOSce) Association.
+#
+
+set name=pkg.fmri value=pkg:/system/library/bhyve@$(PKGVERS)
+set name=pkg.description value="BSD hypervisor (libraries)"
+set name=pkg.summary value="BSD hypervisor (libraries)"
+set name=variant.arch value=i386
+dir path=lib group=bin
+dir path=lib/$(ARCH64) group=bin
+dir path=usr group=sys
+dir path=usr/lib group=bin
+dir path=usr/lib/$(ARCH64) group=bin
+file path=lib/$(ARCH64)/libvmm.so.1
+file path=lib/$(ARCH64)/libvmmapi.so.1
+file path=usr/lib/$(ARCH64)/libppt.so.1
+file path=usr/lib/libppt.so.1
+license lic_CDDL license=lic_CDDL
